### PR TITLE
RHOAIENG-8631: Add image data drift

### PR DESCRIPTION
**Pull request by christinaexyou**
Addresses https://issues.redhat.com/browse/RHOAIENG-8631 --> Add image data drift by making the following changes:
- Detects data drift according to divergences in RGB pixel values

### CODE REVIEW

File: explainability-core/src/main/java/org/kie/trustyai/metrics/drift/image/ImageDrift.java [hunk 1]
```diff
@@ -0,0 +1,102 @@
...
+import java.util.ArrayList;
+import java.util.List;
+
+/**
+ * Jensen-Shannon divergence to calculate image data drift.
+ * See <a href="https://en.wikipedia.org/wiki/Jensen%E2%80%93Shannon_divergence">Jensen–Shannon divergence</a>
+ */
+public class ImageDrift {
+     /**
+     * Converts an image to Dataframe whose columns are "Red", "Green", "Blue" by extracting the associated RGB values.
```
Comment (ruivieira) [inline explainability-core/src/main/java/org/kie/trustyai/metrics/drift/image/ImageDrift.java]: Perhaps we should make a note that alpha channels are ignored?

File: explainability-core/src/main/java/org/kie/trustyai/metrics/drift/image/ImageDrift.java [hunk 2]
```diff
@@ -0,0 +1,102 @@
...
+     *
+     * @param p1 The reference pixel array.
+     * @param p2 The hypothesis pixel array.
+     * @return The Kullback-Leibler divergence.
+     */
+    public static double KLDivergence(double[] p1, double[] p2){
+        double divergence = 0.0;
+        for (int i = 0; i < p1.length; i++){
+            if (p1[i] != 0){
+                divergence += p1[i] * Math.log(p1[i] / p2[i]);
```
Comment (ruivieira) [inline explainability-core/src/main/java/org/kie/trustyai/metrics/drift/image/ImageDrift.java]: What happens when an element of p2 is `0`?
Perhaps catching that case.

File: explainability-core/src/main/java/org/kie/trustyai/metrics/drift/image/ImageDrift.java [hunk 3]
```diff
@@ -0,0 +1,102 @@
...
+    }
+
+     /**
+     * Calculates the Kullback-Leibler divergence between 2 pixel arrays.
+     *
+     * @param p1 The reference pixel array.
+     * @param p2 The hypothesis pixel array.
+     * @return The Kullback-Leibler divergence.
+     */
+    public static double KLDivergence(double[] p1, double[] p2){
```
Comment (ruivieira) [inline explainability-core/src/main/java/org/kie/trustyai/metrics/drift/image/ImageDrift.java]: KLDivergence between two arrays might be useful on its own.
I would move this to a more general utility class for re-use and call that class here.

**Review by ruivieira — CHANGES_REQUESTED** [review]

**Comment by christinaexyou** [comment]
@ruivieira  @RobGeada
Changes made:
`main/.../metrics/drift/image/ImageDrift.java`
- adds support for images of data type `double[][][]`
- converts images of dtype `BufferedImage` to `double[][][]`
- takes as inputs a list of a single image or a multiple images, extending to the data drift and anomaly use cases

`main/.../metrics/drift/utils/KLDivergence.java`
- moved KL Divergence to utils folder

`.tests/.../metrics/drift/utils/KLDivergence.java`
- adds tests for inputs of data type `double[][][]`
- adds a test for mixed input types

**Review by ruivieira — APPROVED** [review]

**Pull request merged**